src/SFML/Window/WindowImpl.hpp

```diff
@@ -282,11 +282,11 @@ class WindowImpl : NonCopyable
     ////////////////////////////////////////////////////////////
     // Member data
     ////////////////////////////////////////////////////////////
-    std::queue<Event> m_events;                                              //!< Queue of available events
-    JoystickState     m_joystickStates[Joystick::Count];                     //!< Previous state of the joysticks
-    Vector3f          m_sensorValue[Sensor::Count];                          //!< Previous value of the sensors
-    float             m_joystickThreshold;                                   //!< Joystick threshold (minimum motion for "move" event to be generated)
-    float             m_previousAxes[Joystick::Count][Joystick::AxisCount];  //!< Position of each axis last time a move event triggered, in range [-100, 100]
+    std::queue<Event>   m_events;                                              //!< Queue of available events
+    JoystickStatesImpl* m_joystickStatesImpl;                                  //!< Previous state of the joysticks (PImpl)
```

Comment:
Is there a reason this can't be a `std::unique_ptr`? As long as we make sure the destructor implementation lives in the `.cpp` the compiler shouldn't complain about incomplete types.

Reply:
I wanted to create a PR that turns every `new`/`delete` into `std::unique_ptr` at once, in order to have a consistent codebase.